pkg/registry/core/pod/strategy_test.go

```diff
@@ -267,53 +268,68 @@ func TestCheckGracefulDelete(t *testing.T) {
 	defaultGracePeriod := int64(30)
 	tcs := []struct {
 		in          *api.Pod
+		out         *metav1.DeleteOptions
 		gracePeriod int64
 	}{
 		{
 			in: &api.Pod{
 				Spec:   api.PodSpec{NodeName: "something"},
 				Status: api.PodStatus{Phase: api.PodPending},
 			},
-
+			out:         &metav1.DeleteOptions{GracePeriodSeconds: &defaultGracePeriod},
 			gracePeriod: defaultGracePeriod,
 		},
 		{
 			in: &api.Pod{
 				Spec:   api.PodSpec{NodeName: "something"},
 				Status: api.PodStatus{Phase: api.PodFailed},
 			},
+			out:         &metav1.DeleteOptions{GracePeriodSeconds: &defaultGracePeriod},
 			gracePeriod: 0,
```

Comment:
if we pass in gracePeriod 0, shouldn't the result be grace period 0, not defaultGracePeriod? (same question applies to the next several test cases)